src/tiledlayer.cpp

```diff
@@ -140,14 +140,14 @@ void TiledLayer::setName(const QString &name)
   \qmlmethod string TiledLayer::getProperty(string name)
   \brief This method returns the value of custom property \e name for this TMX layer.
 */
-QVariant TiledLayer::getProperty(const QString &name) const
+QVariant TiledLayer::getProperty(const QString &name, const QVariant &defaultValue) const
 {
-    if(name.toLower() == "opacity")
+    if(!m_properties.contains(name) && name.toLower() == "opacity")
         return QVariant::fromValue(m_layer->opacity());
-    else if(name.toLower() == "visible")
+    else if(!m_properties.contains(name) && name.toLower() == "visible")
```

Comment:
I would suggest to do the contains check once and do case-insensitive comparison.
```c++
if (!m_properties.contains(name)) {
    if (name.compare("opacity", Qt::CaseInsensitive) == 0)
        return QVariant::fromValue(m_layer->opacity());
    ...
}
```
(this especially affects `TiledObject::getProperty`)

Though, I have to say that I don't understand why a case-insensitive check is used in the first place, but the code was already like that.